extensions/positron-run-app/src/api.ts

```diff
@@ -353,18 +353,27 @@ async function previewUrlInExecutionOutput(execution: vscode.TerminalShellExecut
 		return false;
 	}
 
-	// Convert the url to an external URI.
+	// Example: http://localhost:8500
 	const localBaseUri = vscode.Uri.parse(url.toString());
+
+	// Example: http://localhost:8500/url/path or http://localhost:8500
 	const localUri = urlPath ?
 		vscode.Uri.joinPath(localBaseUri, urlPath) : localBaseUri;
 
-	log.debug(`Viewing app at local uri: ${localUri} with external uri ${proxyInfo.externalUri.toString()}`);
+	// Determine the target origin based on whether the server is running in PWB.
+	const runningInPWB = process.env.RS_SERVER_URL ? true : false;
+	const targetOrigin = runningInPWB ? localBaseUri : localUri;
```

Comment:
Interesting. I'm curious why the target origin differs here for workbench

Reply:
It's a bit of a hack as I couldn't get `UVICORN_ROOT_PATH` or `--root-path` to work for fastapi in Positron Server Web or Desktop. I haven't gotten to the bottom of why it's working on Workbench.

The result on Positron Server Web without this handling is that the `openapi.json` file is getting loaded from the root `/openapi.json` instead of from e.g. `proxy/1234/openapi.json`, so we get:

<img width="1208" alt="positron_web_fastapi_openapi_notfound" src="https://github.com/user-attachments/assets/93e9cbd2-96bc-441f-8388-f18fc564d347">

Reply:
I'm gonna move this PR to Draft while I try a few more things to see if we can avoid this custom handling.